### PR TITLE
[SPARK-38778][INFRA][BUILD] Replace http with https for project url in pom

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

change <url>http://spark.apache.org/</url> to <url>https://spark.apache.org/</url> in the project URL of all pom files
### Why are the changes needed?

fix home page in maven central https://mvnrepository.com/artifact/org.apache.spark/spark-sql_2.13/3.2.1

#### From
License | Apache 2.0
-- | --
Categories |Hadoop Query Engines
HomePage|http://spark.apache.org/
Date | (Jan 26, 2022)

#### to

License | Apache 2.0
-- | --
Categories |Hadoop Query Engines
HomePage|https://spark.apache.org/
Date | (Jan 26, 2022)
### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

pass GHA

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
   <version>3.4.0-SNAPSHOT</version>
   <packaging>pom</packaging>
   <name>Spark Project Parent POM</name>
-  <url>http://spark.apache.org/</url>
```
Comment (wangyum) [inline pom.xml:32]: Line 53?
https://github.com/apache/spark/blob/629187e767d84f55fd849b0e8f09ad05a6f3f139/pom.xml#L53

Comment (yaooqinn) [reply pom.xml:32]: updated, thanks.

**Review by wangyum — APPROVED** [review]

**Comment by wangyum** [comment]
Merged to master and branch-3.3.

**Comment by yaooqinn** [comment]
thanks @HyukjinKwon @wangyum for the review

**Pull request closed without merging**